packages/api/tests/app/test-auth.js

```diff
@@ -0,0 +1,192 @@
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Authorization scheme must be Bearer');
+    });
+});
+
+test('ensureAuthorized() returns an unauthorized response if the token is missing', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', 'Bearer')
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Missing token');
+    });
+});
+
+test.serial('ensureAuthorized() verifies that the user is authorized when not using Launchpad authentication', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  process.env.OAUTH_PROVIDER = 'earthdata';
+
+  const jwtToken = createJwtToken();
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', `Bearer ${jwtToken}`)
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'User not authorized');
+    });
+});
+
+test.serial('If Launchpad authentication is enabled, and the Bearer token is not a valid JWT token, ensureAuthorized() passes the authentication over to ensureLaunchpadAPIAuthorized()', async (t) => {
+  const { ensureAuthorizedApp } = t.context;
+
+  const launchpadToken = randomString();
+  const expectedAuthorizationToken = `Bearer ${launchpadToken}`;
+
+  process.env.OAUTH_PROVIDER = 'launchpad';
+
+  let wasEnsureLaunchpadAPIAuthorizedCalled = false;
+
+  await auth.__with__({
+    ensureLaunchpadAPIAuthorized: async (req, _res, next) => {
+      t.is(req.headers.authorization, expectedAuthorizationToken);
+      wasEnsureLaunchpadAPIAuthorizedCalled = true;
+      return next();
+    }
+  })(
+    () =>
+      request(ensureAuthorizedApp)
+        .get('/asdf')
+        .set('Authorization', expectedAuthorizationToken)
+        .then((res) => {
+          t.is(res.statusCode, 200);
+          t.true(wasEnsureLaunchpadAPIAuthorizedCalled);
+        })
+  );
+});
+
+test.serial('ensureAuthorized() returns unauthorized if the token has expired', async (t) => {
+  const { accessTokenModel, ensureAuthorizedApp, userModel } = t.context;
+
+  const accessToken = fakeAccessTokenFactory({
+    expirationTime: Date.now() - (10 * 1000)
+  });
+  await accessTokenModel.create(accessToken);
+
+  const jwtToken = createJwtToken(accessToken);
+
+  await userModel.create({ userName: accessToken.username });
+
+  process.env.OAUTH_PROVIDER = 'earthdata';
+
+  await request(ensureAuthorizedApp)
+    .get('/asdf')
+    .set('Authorization', `Bearer ${jwtToken}`)
+    .then((res) => {
+      t.is(res.statusCode, 401);
+      const body = JSON.parse(res.text);
+      t.is(body.message, 'Access token has expired');
+    });
+});
+
+test.serial('If Launchpad authentication is not enabled, and the Bearer token is not a valid JWT token, ensureAuthorized() returns a forbidden response', async (t) => {
```

Comment:
I think I'd like to see one more test here where Launchpad authentication is enabled and the token is made invalid by using an invalid secret. You can achieve this easily by setting a value for `process.env.TOKEN_SECRET` before creating the token and then changing that value before testing the endpoint, so that the two secrets won't match. This ensures that the Launchpad auth scenario still verifies JWT validity in cases where it is not a "malformed" token that is passed off to `ensureLaunchpadAPIAuthorized`